src/extension/app-schema/app-schema-mongo-test/src/test/java/org/geoserver/test/onlineTest/ComplexMongoDBTest.java

```diff
@@ -0,0 +1,273 @@
+package org.geoserver.test.onlineTest;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.DBObject;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.ServerAddress;
+import com.mongodb.util.JSON;
+import org.custommonkey.xmlunit.Diff;
+import org.custommonkey.xmlunit.Difference;
+import org.custommonkey.xmlunit.DifferenceListener;
+import org.custommonkey.xmlunit.examples.RecursiveElementNameAndTextQualifier;
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.CatalogBuilder;
+import org.geoserver.catalog.FeatureTypeInfo;
+import org.geoserver.catalog.LayerInfo;
+import org.geoserver.catalog.impl.DataStoreInfoImpl;
+import org.geoserver.catalog.impl.NamespaceInfoImpl;
+import org.geoserver.catalog.impl.WorkspaceInfoImpl;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.geoserver.util.IOUtils;
+import org.geotools.feature.NameImpl;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.logging.Logger;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assume.assumeThat;
+
+/**
+ * Tests the integration between MongoDB and App-schema. This test are integration
+ * tests hence they require a MongoDB instance. Host 127.0.0.1 and port 27017 will
+ * be used by default but is possible to provide another host and port using Java
+ * variables mongo.host and mongo.port respectively.
+ */
+public final class ComplexMongoDBTest extends GeoServerSystemTestSupport {
+
+    private static final Logger LOGGER = org.geotools.util.logging.Logging.getLogger(ComplexMongoDBTest.class);
+
+    private static final Path ROOT_DIRECTORY = createTempDir();
+
+    private static File APP_SCHEMA_MAPPINGS;
+
+    private static final String STATIONS_DATA_BASE_NAME = UUID.randomUUID().toString();
+    private static final String STATIONS_COLLECTION_NAME = "stations";
+
+    private static MongoClient MONGO_CLIENT;
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        // check that we have access to a mongodb and instantiate the client
+        String hostAsString = getProperty("mongo.host", "127.0.0.1");
```

Comment:
This test fixture should use settings stored in a properties file ```~/.geoserver/mongodb.properties``` or similar. GeoTools fixture configuration support is more reusable, but GeoServer is messier. ```ReferenceDataPostgisSetup``` uses ```JDBCTestSetup``` but I expect that this will be less useful for Mongo. GeoTools MongoTestSupport does the right thing.